tests/scenario_tests/test_events_shared_channels.py

```diff
@@ -0,0 +1,514 @@
+        "token": "verification_token",
+        "team_id": "T_INSTALLED",
+        "enterprise_id": "E_INSTALLED",
+        "api_app_id": "A111",
+        "event": {
+            "client_msg_id": "9cbd4c5b-7ddf-4ede-b479-ad21fca66d63",
+            "type": "app_mention",
+            "text": "<@W111> Hi there!",
+            "user": "W222",
+            "ts": "1595926230.009600",
+            "team": "T_INSTALLED",
+            "channel": "C111",
+            "event_ts": "1595926230.009600",
+        },
+        "type": "event_callback",
+        "event_id": "Ev111",
+        "event_time": 1595926230,
+        "authorizations": [
+            {
+                "enterprise_id": "E_INSTALLED",
+                "team_id": "T_INSTALLED",
+                "user_id": "W111",
+                "is_bot": True,
+                "is_enterprise_install": False,
+            }
+        ],
+    }
+
+    def test_mock_server_is_running(self):
+        resp = self.web_client.api_test(token=valid_token)
+        assert resp != None
+
+    def test_middleware(self):
+        app = App(
+            client=self.web_client,
+            signing_secret=self.signing_secret,
+            authorize=authorize,
+        )
+
+        @app.event("app_mention")
+        def handle_app_mention(body, say: Say, payload, event):
+            assert body == self.valid_event_body
+            assert body["event"] == payload
+            assert payload == event
+            say("What's up?")
+
+        timestamp, body = str(int(time())), json.dumps(self.valid_event_body)
+        request: BoltRequest = BoltRequest(
+            body=body, headers=self.build_headers(timestamp, body)
+        )
+        response = app.dispatch(request)
+        assert response.status == 200
+        assert self.mock_received_requests["/auth.test"] == 1
+        sleep(1)  # wait a bit after auto ack()
+        assert self.mock_received_requests["/chat.postMessage"] == 1
+
+    def test_middleware_skip(self):
+        app = App(
+            client=self.web_client,
+            signing_secret=self.signing_secret,
+            authorize=authorize,
+        )
+
+        def skip_middleware(req, resp, next):
+            # return next()
+            pass
+
+        @app.event("app_mention", middleware=[skip_middleware])
+        def handle_app_mention(body, logger, payload, event):
+            assert body["event"] == payload
+            assert payload == event
+            logger.info(payload)
+
+        timestamp, body = str(int(time())), json.dumps(self.valid_event_body)
+        request: BoltRequest = BoltRequest(
+            body=body, headers=self.build_headers(timestamp, body)
+        )
+        response = app.dispatch(request)
+        assert response.status == 404
+        assert self.mock_received_requests["/auth.test"] == 1
+
+    valid_reaction_added_body = {
+        "token": "verification_token",
+        "team_id": "T_SOURCE",
+        "enterprise_id": "E_SOURCE",
```

Comment:
We used to us these values but they may not be the ones we need when an app works in shared channels.